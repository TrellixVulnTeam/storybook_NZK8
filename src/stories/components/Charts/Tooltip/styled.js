// @flow
import styled from 'styled-components';

export const TooltipWrapper = styled.div`
  background-color: rgba(54, 56, 57, 0.8);
  color: #fff;
  padding: 10px 15px;
  border-radius: 4px;
  font-size: 14px;
  font-weight: 600;
`;

export const Label = styled.span``;

export const Value = styled.span`
  font-style: italic;
  margin-left: 4px;
`;
export const Percent = styled.span``;
