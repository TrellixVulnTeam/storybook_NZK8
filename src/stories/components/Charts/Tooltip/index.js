// @flow
/* eslint-disable import/prefer-default-export */
import React from 'react';
import { TooltipWrapper, Label, Value, Percent } from './styled';

/**
 * renders Tooltip item
 * @param {Object} props
 * @return {Node}
 */
export const renderTooltip = (props: Object) => {
  const { payload } = props;
  return payload.length ? (
    <TooltipWrapper>
      <Label>{payload[0].payload.name}</Label> :<Percent>{payload[0].payload.value}</Percent>%
      <Value>({payload[0].payload.actualValue})</Value>
    </TooltipWrapper>
  ) : (
    ''
  );
};
