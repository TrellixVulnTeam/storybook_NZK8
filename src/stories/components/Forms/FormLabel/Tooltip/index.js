// @flow

import React from 'react';
import TooltipTrigger from 'react-popper-tooltip';
import { TooltipPlaceholder, TooltipWrapper } from './styled';

/**
 * @param {string[] | string} messages
 * @return {Function}
 */
const getTooltipContent = (messages: string[] | string) => ({
  arrowRef,
  tooltipRef,
  getArrowProps,
  getTooltipProps,
  placement,
}) => {
  return (
    <TooltipWrapper>
      <div
        {...getTooltipProps({
          ref: tooltipRef,
          className: 'tooltip-container',
        })}
      >
        <div
          {...getArrowProps({
            ref: arrowRef,
            className: 'tooltip-arrow',
            'data-placement': placement,
          })}
        />
        {Array.isArray(messages) ? (
          <ul className="tooltip-body">
            {messages.map((message, i) => (
              // eslint-disable-next-line react/no-array-index-key
              <li key={i}>{message}</li>
            ))}
          </ul>
        ) : (
          <div className="tooltip-body tooltip-body__single-message">{messages}</div>
        )}
      </div>
    </TooltipWrapper>
  );
};

type TriggerProps = {
  getTriggerProps: Function,
  triggerRef: React$ElementRef<any>,
};

/**
 * @param {TriggerProps} props
 * @return {React$Node}
 */
const Trigger = ({ getTriggerProps, triggerRef }: TriggerProps) => {
  return (
    <span
      {...getTriggerProps({
        ref: triggerRef,
        className: 'trigger',
      })}
    >
      <TooltipPlaceholder />
    </span>
  );
};

/**
 * @param {string[] | string} messages
 * @return {React$Node}
 */
const Tooltip = ({ messages }: { messages: string[] | string }) => {
  return (
    <TooltipTrigger placement="right-start" trigger="hover" tooltip={getTooltipContent(messages)}>
      {Trigger}
    </TooltipTrigger>
  );
};

export default Tooltip;
