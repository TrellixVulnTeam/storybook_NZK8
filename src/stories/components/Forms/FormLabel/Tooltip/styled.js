// @flow

import styled from 'styled-components';
import { colors } from 'utils/StyleGuide';

export const TooltipPlaceholder = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 18px;
  height: 18px;
  margin: 0 5px;
  user-select: none;
  color: ${colors.white};
  background-color: ${colors.steel};
  border-radius: 50%;

  :after {
    content: '?';
  }
`;

export const TooltipWrapper = styled.div`
  .tooltip-container {
    width: 365px;
    top: -15px !important;
    color: ${colors.white};
    background-color: ${colors.darkBlue};
    font-size: 14px;
    border: none;
  }

  .tooltip-body {
    margin: 5px 20px 5px 30px;
    padding: 0;
    font-family: 'Open Sans', sans-serif;
    line-height: 1.7;

    li {
      margin-bottom: 5px;
    }

    &__single-message {
      margin: 5px 20px 5px 10px;
    }
  }

  .tooltip-arrow {
    top: 10px !important;

    :after {
      border-color: transparent ${colors.darkBlue} transparent transparent !important;
    }

    :before {
      border-color: transparent transparent transparent ${colors.darkBlue} !important;
    }
  }
`;
