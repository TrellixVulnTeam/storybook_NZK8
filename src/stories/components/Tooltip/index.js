// @flow

import React from 'react';
import ReactTooltip from 'react-tooltip';
import { Wrapper, TooltipWrapper } from './styled';

// eslint-disable-next-line no-unused-vars
type Props = {
  children: React$Node,
  id: String,
  value: String,
  display?: 'true' | 'false',
  [key: string]: any,
};

export default (props: $FlowLintFix) => (
  <Wrapper data-for={props.id} data-tip={props.value} {...props}>
    {props.children}
    <TooltipWrapper display={props.display}>
      <ReactTooltip
        id={props.id}
        html={props.html || true}
        getContent={dataTip => `${dataTip}`}
        type={props.type || 'dark'}
        place={props.place || 'top'}
        multiline={props.multiline || true}
        delayShow={props.delayShow || 300}
        effect={props.effect || 'solid'}
      />
    </TooltipWrapper>
  </Wrapper>
);
