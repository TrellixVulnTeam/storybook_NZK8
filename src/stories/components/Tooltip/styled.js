// @flow

import styled from 'styled-components';
import { colors } from 'utils/StyleGuide';

export const Wrapper = styled.div`
  display: inline-flex;
`;

export const TooltipWrapper = styled.div`
  display: ${p => (JSON.parse(p.display) ? 'block' : 'none')};

  .__react_component_tooltip {
    font-size: 12px;
    padding: 8px;
    background-color: ${colors.darkGrey};
    box-shadow: 0 2px 20px 0 rgba(0, 0, 0, 0.5);
  }

  aside {
    display: block;
    height: 10px;
  }

  i {
    display: block;
    font-size: 10px;
    color: ${colors.silver};
  }
`;

TooltipWrapper.defaultProps = {
  display: 'true',
};
