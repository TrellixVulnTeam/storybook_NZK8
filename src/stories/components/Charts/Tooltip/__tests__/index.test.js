import { renderTooltip } from 'components/Charts/Tooltip/index';

describe('<renderTooltip />', () => {
  const props = {
    payload: [
      {
        payload: {
          name: 'Name',
          value: 0,
          actualValue: 0,
        },
      },
    ],
  };

  it('should render correctly with props', () => {
    const sut = renderTooltip(props);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly with empty payload', () => {
    const sut = renderTooltip({ payload: [] });

    expect(sut).toMatchSnapshot();
  });
});
