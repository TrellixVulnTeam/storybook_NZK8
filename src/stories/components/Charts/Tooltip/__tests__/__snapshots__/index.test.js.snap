// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<renderTooltip /> should render correctly with empty payload 1`] = `""`;

exports[`<renderTooltip /> should render correctly with props 1`] = `
<styled__TooltipWrapper>
  <styled__Label>
    Name
  </styled__Label>
   :
  <styled__Percent>
    0
  </styled__Percent>
  %
  <styled__Value>
    (
    0
    )
  </styled__Value>
</styled__TooltipWrapper>
`;
